core/src/test/java/org/apache/calcite/test/enumerable/EnumerableBatchNestedLoopJoinTest.java

```diff
@@ -226,6 +226,7 @@ class EnumerableBatchNestedLoopJoinTest {
             + "join locations l on e.empid <> l.empid and d.deptno = l.empid")
         .withHook(Hook.PLANNER, (Consumer<RelOptPlanner>) planner -> {
           planner.removeRule(EnumerableRules.ENUMERABLE_CORRELATE_RULE);
+          planner.removeRule(EnumerableRules.ENUMERABLE_JOIN_RULE);
```

Comment:
is this related to the problem addressed?

Reply:
It's related to the test. When the rule is not removed, the planner creates a `EnumerableHashJoin` instead of a `EnumerableBatchNestedLoopJoin` because the cost function is affected by this change.